drivers/reset/reset_stm32mp1.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32mp1_rcc_rctl
```

Comment:
Instead of duplicating the whole driver, can we think about a (n optional) property than could be use to differentiate behaviors?

Reply:
Yes, it's possible. I would change `reset_stm32_line_deassert()` to support STM32MP1 + some define changes in driver. Do you think it would be cleaner? Do you have a name for the property?

PS I found bug in drivers/reset/reset_stm32mp1.c just now, so this is not final version.

Reply:
One idea would be to introduce 2 props on all series which would be set-offset/clr-offset with value 5/5 on most series except MP1 where it would be 5/17.

Reply:
To deassert reset on STM32MP1 you need to set but in clear register (clearing bit in set/clear register does nothing), so we actually need 3 properties. What do you think about it? Can I use default value for set-offset/clr-offset, and add these properties in stm32mp1.dtsi?

Reply:
Alternative is to use the following for MP1:
```
         compatible = "st,stm32mp1-rcc-rctl", "st,stm32-rcc-rctl";
```
This way you get both compatibles defined when compiled for MP1 case.
Then you can use `DT_HAS_COMPAT_ENABLED(st_stm32mp1_rcc_rct)` to tweak stm32 reset driver behavior in MP1 case. Similar trick is used in SPI and I2C STM32 drivers.

Reply:
I added `set-bit-to-deassert` property. It's used in MP1 `rctl` node to indicate that the driver should set bit in clear register to deassert reset.